core/src/impl/Kokkos_AnalyzePolicy.hpp

```diff
@@ -178,11 +202,18 @@ struct AnalyzePolicy<Base>
                                                      , void // TODO set default iteration pattern
                                                      , typename Base::iteration_pattern
                                                      >::type;
+
+  using launch_bounds = typename std::conditional< is_void< typename Base::launch_bounds >::value
+                                                     , LaunchBounds<1024,1> // TODO set correct bounds for each device
```

Comment:
What is the impact on performance of having the incorrect default launch bounds?  Would it be better to have no launch bounds instead of incorrect launch bounds?  A default of `void` could be propagated to the launch such that the absence of specified launch bounds results in omission of the launch bounds clause.

Reply:
Somehow this default, even though it is maximally permissive, seems to give much worse performance than not specifying the launch bounds at all. See #1139.

Reply:
Thanks for catching this, Stan, and sorry about any trouble.

Pull request 1143 should fix (https://github.com/kokkos/kokkos/pull/1143).

Do you, Stan, have a good test case to demonstrate the slow down?

Reply:
@LeviBarnes yes, I am using the LAMMPS Tersoff potential. I put a description of how to reproduce the slow down in #1139, you would need to build Kokkos-enabled LAMMPS.